app/league/[leagueId]/entry/[entryId]/week/Week.test.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import Alert from '@/components/AlertNotification/AlertNotification';
+import { AlertVariants } from '@/components/AlertNotification/Alerts.enum';
+import { toast } from 'react-hot-toast';
+import { onWeeklyPickChange } from './WeekHelper';
+import { createWeeklyPicks } from '../../../../../../api/apiFunctions';
+import { parseUserPick } from '../../../../../../utils/utils';
+
+jest.mock('../../../../../../api/apiFunctions', () => ({
+  createWeeklyPicks: jest.fn(),
+}));
+
+jest.mock('react-hot-toast', () => ({
+  toast: {
+    custom: jest.fn(),
+  },
+}));
+
+describe('Week', () => {
+  const data = {
+    target: { value: 'Browns' },
+    preventDefault: jest.fn(),
+    stopPropagation: jest.fn(),
+  } as any;
+  const NFLTeams = [{ teamName: 'Browns', teamId: 'mockId' }];
+  const user = { id: 'mockUserId', email: 'email@example.com', leagues: [] };
+  const entry = 'mockEntry';
+  const league = 'mockLeague';
+  const week = 'mockWeek';
+  const updateWeeklyPicks = jest.fn();
+  const setUserPick = jest.fn();
+  interface IUserResults {
+    [key: string]: any; // Adjust the 'any' to match the structure of your user results
+  }
+
+  interface IWeeklyPicks {
```

Comment:
Why are we recreating the interfaces in a test?

Reply:
I had to create mock ones to define everything else in the tests

Reply:
oh sorry i understand now and fixed it. I keep thinking that in tests you have to explain EVERYTHING to it. I didn't realize you can easily import IWeeklyPicks and use it inside the test easily